app.rb

```diff
@@ -0,0 +1,132 @@
+require './person'
+require './student'
+require './teacher'
+require './book'
+require './classroom'
+require './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def start_console
+    puts 'Welcome to School Library!'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank You for using School Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_of_books
+    puts 'Empty! Please add a book.' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_of_persons
+    puts 'Empty! Please add a person.' if @persons.empty?
+    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}" }
+  end
+
+  def create_person
+    print 'To create a student, press 1, To create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new('classroom', age, name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new('classroom', age, name, parent_permission: true)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Enter teacher age: '
+    age = gets.chomp.to_i
+    print 'Enter teacher name: '
+    name = gets.chomp
+    print 'Enter teacher specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @persons << teacher
+    puts 'Teacher created successfully'
+  end
+
+  def create_book()
+    puts 'Create a new book'
+    print 'Enter title: '
+    title = gets.chomp
+    print 'Enter author: '
+    author = gets
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book #{title} created successfully."
+  end
+
+  def create_rental
+    puts 'Select which book you want to rent by entering its number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the list by its number'
+    @persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+
```

Comment:
- [ ] As requested by the previous code reviewer, which is a requirement for this project, please I am unable to list the rentals for a give id, I am getting this message " `No records where found for the given ID` ". Please checkout the screenshot below and kindly fix the issue.

![Screen Shot 2022-04-22 at 04 40 54](https://user-images.githubusercontent.com/62849872/164699514-33755389-4cc8-44b4-b947-0551bcdadac6.png)
.

Reply:
- [ ] As requested by the previous reviewers, the issue still exists for listing the rental given an ID. Please fix it as suggested you will move to the next milestone. 

![image](https://user-images.githubusercontent.com/55635977/164737410-e5b654c5-4d16-4cb2-ba7b-5874c510d41c.png)